test/e2e/installplan_e2e_test.go

```diff
@@ -2869,92 +2869,127 @@ var _ = Describe("Install Plan", func() {
-					if !ok {
-						errExit <- errors.New("cr watch channel closed unexpectedly")
-						return
-					}
-					if evt.Type == watch.Deleted {
-						cr, ok := evt.Object.(*rbacv1.ClusterRole)
+		GinkgoT().Logf("deleting csv %s/%s", generatedNamespace.GetName(), stableCSVName)
+		Eventually(func(g Gomega) bool {
+			done := make(chan struct{})
+			errExit := make(chan error)
+			go func() {
+				defer GinkgoRecover()
+				for {
+					select {
+					case evt, ok := <-saWatcher.ResultChan():
 						if !ok {
-							continue
-						}
-						delete(createdClusterRoleNames, cr.GetName())
-						if len(createdClusterRoleNames) == 0 && len(createdClusterRoleBindingNames) == 0 {
-							done <- struct{}{}
+							errExit <- errors.New("sa watch channel closed unexpectedly")
 							return
 						}
-					}
-				case evt, ok := <-crbWatcher.ResultChan():
-					if !ok {
-						errExit <- errors.New("crb watch channel closed unexpectedly")
-						return
-					}
-					if evt.Type == watch.Deleted {
-						crb, ok := evt.Object.(*rbacv1.ClusterRoleBinding)
+						if evt.Type == watch.Deleted {
+							sa, ok := evt.Object.(*corev1.ServiceAccount)
+							if !ok {
+								continue
+							}
+							GinkgoT().Logf("sa %v deleted", sa.GetName())
+							delete(createdResources, "sa/"+sa.GetName())
+							if len(createdResources) == 0 {
+								done <- struct{}{}
+								return
+							}
+						}
+					case evt, ok := <-crWatcher.ResultChan():
 						if !ok {
-							continue
+							errExit <- errors.New("cr watch channel closed unexpectedly")
+							return
+						}
+						if evt.Type == watch.Deleted {
+							cr, ok := evt.Object.(*rbacv1.ClusterRole)
+							if !ok {
+								continue
+							}
+							GinkgoT().Logf("cr %v deleted", cr.GetName())
+							delete(createdResources, "cr/"+cr.GetName())
+							if len(createdResources) == 0 {
+								done <- struct{}{}
+								return
+							}
 						}
-						delete(createdClusterRoleBindingNames, crb.GetName())
-						if len(createdClusterRoleNames) == 0 && len(createdClusterRoleBindingNames) == 0 {
-							done <- struct{}{}
+					case evt, ok := <-crbWatcher.ResultChan():
+						if !ok {
+							errExit <- errors.New("crb watch channel closed unexpectedly")
 							return
 						}
+						if evt.Type == watch.Deleted {
+							crb, ok := evt.Object.(*rbacv1.ClusterRoleBinding)
+							if !ok {
+								continue
+							}
+							GinkgoT().Logf("crb %v deleted", crb.GetName())
+							delete(createdResources, "crb/"+crb.GetName())
+							if len(createdResources) == 0 {
+								done <- struct{}{}
+								return
+							}
+						}
+					case <-time.After(pollDuration):
+						done <- struct{}{}
+						return
 					}
-				case <-time.After(pollDuration):
-					done <- struct{}{}
-					return
 				}
+			}()
+			err = crc.OperatorsV1alpha1().ClusterServiceVersions(generatedNamespace.GetName()).DeleteCollection(context.Background(), metav1.DeleteOptions{}, metav1.ListOptions{})
```

Comment:
Without selectors on the ListOptions I don't think this call will do anything.

Reply:
Hmmm... this is basically the same call as in the original code.

Reply:
Which is probably why I added an explicit delete... AFAICT, it was a delete of all CSVs in the `generatedNamespace()`?

Reply:
Huh - you're right. No selector => every namespaced object. Wild. Rarely-used corner of the k8s API.

Reply:
I removed it, so that I'm now targeting the specific CSV.